src/internal/connector/discovery/discovery.go

```diff
@@ -69,6 +69,22 @@ func Users(
 	return users, nil
 }
 
+// UserDetails fetches detailed info like - userPurpose for all users in the tenant.
+func UserDetails(
```

Comment:
Nit: since we're getting user info
```suggestion
func GetUserInfo(
```